### DISCUSSION TITLE
Proxy switching

### DISCUSSION THREAD

**Discussion by lumotroph** [snappysnappydog/Simple-IP-Config#115] [Ideas]
I'm at a university where the **wifi** connection goes through a **proxy** - of course I switch the proxy settings off (to "automatic detection") whenever i'm not on campus.

This awesome program would be a great further help if the proxy settings could be toggled too!

**Comment by KurtisLiggett** [comment]
Can you tell me how you currently set your proxy settings?

**Comment by lumotroph** [comment]
So currently I press the windows key, type "inter..." and *Internet options* comes up, which I open.

Then under "Connections" there's a "LAN Settings" section, under which are two checkboxes which I switch depending on the network - "Automatically detect settings" on one of them, and "Use a proxy server for your LAN..." on the other one.


![proxy-settings](https://user-images.githubusercontent.com/12592541/36279422-bfde1bc8-129f-11e8-92cb-02c15d20945d.gif)

^ this process is done in reverse later.

**Comment by controlzone** [comment]
+1 This would be fantastic - when I am in the office I need to have the proxy but most other sites dont have one and the autodetect isnt very fast / accurate. Some other sites have their own proxy also.

**Comment by Melog01** [comment]
+1 It will help me as our logon script is setting it to automatically and that setting is not working with some software. Please do not forget the Advanced proxy setting as I have to do it manually now (e.g. different proxy's for different type and more exceptions). 

![image](https://user-images.githubusercontent.com/22768871/53969404-09629200-40f9-11e9-8265-abbf0cf2d248.png)

Thank you

**Comment by decadenza** [comment]
I think that this feature would make this the definitive tool!

**Comment by KurtisLiggett** [comment]
There seems to be some growing interest in this, so I will check out what it would take to add.
I have some big changes coming up which might make it easier to add something like this.
